plasmapy/utils/code_repr.py

```diff
@@ -0,0 +1,321 @@
+"""Tools for formatting strings, including for error messages."""
+
+__all__ = ["call_string", "attribute_call_string", "method_call_string"]
+
+import inspect
+import numpy as np
+
+from astropy import units as u
+from typing import Any, Callable, Dict, List, Optional, Tuple, Union
+
+
+def _code_repr_of_ndarray(array: np.ndarray, max_items=np.inf) -> str:
+    """
+    Transform a `~numpy.ndarray` into a format as would appear in a
+    function call, after having done ``import numpy as np``.
+
+    If ``max_items`` is less than ``array.size``, then include only the
+    first ``max_items`` elements of the array in the resulting string
+    and replace the remaining items with ``"..."``.
+    """
+
+    def remove_excess_spaces(string: str) -> str:
+        string = " ".join(string.split())
+        string = string.replace(" ,", ",")
+        string = string.replace("[ ", "[")
+        return string
+
+    def put_np_before_infs_and_nans(string: str) -> str:
+        string = string.replace("inf", "np.inf")
+        string = string.replace("nan", "np.nan")
+        return string
+
+    def replace_excess_items_with_ellipsis(s, max_items):
+        substrings_between_commas = s.split(",")
+        to_comma_before_ellipsis = ",".join(substrings_between_commas[0:max_items])
+        closing_brackets = "]" * substrings_between_commas[-1].count("]")
+        closing = f", ...{closing_brackets})"
+        return to_comma_before_ellipsis + closing
+
+    if not isinstance(array, np.ndarray):
+        raise TypeError("Expecting an ndarray.")
```

Comment:
This scenario is not covered in the tests.

Reply:
Since these are private functions, it might be better to do EAFP rather than LBYL (following a code review comment I received on Astropy).  I think I'll delete these lines.